### PR TITLE
Add client-side functionality to archive tests

### PR DESCRIPTION
**Pull request by tjsilver**
Also connected client-side functionality to archive endpoint.

### Add archive button next to delete button (only in edit mode): 
image

### Highlight archived tests in list:
image

### Show number of archived tests in save popover:
image

### Show to be archived in heading:
image

### CODE REVIEW

File: public/src/components/epicTests/epicTestEditor.tsx [hunk 1]
```diff
@@ -22,35 +22,36 @@ import EditableTextField from "../helpers/editableTextField"
...
+    margin: '10px 0 15px 0'
   },
   hasChanged: {
     color: 'orange'
   },
   h4: {
     fontSize: typography.pxToRem(24),
     fontWeight: typography.fontWeightMedium,
-    margin: "20px 0 15px 0"
+    margin: '20px 0 15px 0'
```
Comment (jlieb10) [inline public/src/components/epicTests/epicTestEditor.tsx]: if you prefer being succinct, this is equivalent to
```suggestion
    margin: '20px 0 15px'
```

File: public/src/components/epicTests/epicTestEditor.tsx [hunk 2]
```diff
@@ -22,35 +22,36 @@ import EditableTextField from "../helpers/editableTextField"
...
     marginTop: spacing(2),
     marginBottom: spacing(1),
-    display: "block",
+    display: 'block',
   },
   h3: {
     fontSize: typography.pxToRem(28),
     fontWeight: typography.fontWeightMedium,
-    margin: "10px 0 15px 0"
+    margin: '10px 0 15px 0'
```
[the remark this answers is not in the document]
Comment (tjsilver) [reply public/src/components/epicTests/epicTestEditor.tsx]: Good spot!

File: public/src/components/epicTests/epicTestEditor.tsx (hunk 1 above)
Comment (tjsilver) [reply public/src/components/epicTests/epicTestEditor.tsx]: 👍

File: public/src/components/epicTests/epicTestEditor.tsx [hunk 3]
```diff
@@ -109,15 +119,17 @@ class EpicTestEditor extends React.Component<EpicTestEditorProps, EpicTestEditor
     validationStatus: {}
   };
 
+  areYouSure = `Are you sure? This can't be undone without cancelling entire edit session!`;
```
[the remark this answers is not in the document]
Comment (tjsilver) [reply public/src/components/epicTests/epicTestEditor.tsx]: Good idea!

**Comment by tjsilver** [comment]
> Just being devil's advocate here- what's the advantage of allowing there to be delete functionality and archive functionality? Should we get rid of delete and only offer archive, like Trello does?

At present, delete allows a user to delete a test that was created in error, say with an incorrect name.

**Pull request merged**